conformance/README.md

```diff
@@ -561,9 +561,11 @@ type.
 These rules describe the overall shape of test cases:
 
 ```ebnf
-test ::=  "("  "ion_1_0"  fragment*  continuation  ")"
-       |  "("  "ion_1_1"  fragment*  continuation  ")"
-       |  "("  "ion_1_x"  fragment*  continuation  ")"
+test ::=  "("  "ion_1_0"  name-string  fragment*  continuation  ")"
+       |  "("  "ion_1_1"  name-string  fragment*  continuation  ")"
+       |  "("  "ion_1_x"  name-string  fragment*  continuation  ")"
```

Comment:
I think these name-strings should be optional as they are in the extensions.

(I'm working on a PR making it more explicit that the `ion_` forms are shorthands for `each` clauses, so I think they should behave equivalently.)

Reply:
As a reviewer, I will ask for a `name-string` (or `description` or whatever we call it) for every test case that I review. I would prefer that the grammar enforce that, but I think it could be okay to call it optional if that makes it significantly easier to implement the test DSL.

That being said, I also don't mind if the runners don't perfectly implement the grammar. For example, if the grammar says that the name/description is required, but an implementation doesn't actually enforce that, that's not a big deal to me. Test runners should be able to assume that the tests are well formed.

I'm planning to (eventually) write ISL for the conformance DSL, and then we can have a CI check to validate the tests as we do with `ion-schema-tests`.

Reply:
On the whole, I think requiring the text is creating low-value busy work. Many test trees are just variations on a theme, and fairly obvious.  In practice it's fairly difficult to produce meaningful, readable composite descriptions for every branch on the test tree.

I see this text as just a comment that passes through the test framework to developers so that when a test fails they can have some notion of what broke.

I'm looking at a the tests covering `for`, which I rewrote yesterday into a single `ion_1_1` clause as kind of a stress test.  It covers about 100 LOC, has about 30 then/else clauses nested up to three levels deep.  Many of those branches are just covering fairly obvious yet repetitive combinations of things.  I don't think it would be worth effort to annotate each and every one, much less compose them such that combine into readable sentences.  At present, I have labels at the top and at most one more level, going further is not worth my time.

Reply:
Sorry I meant to also say that I'm not particularly opposed to requiring one in the outermost clauses.  Perhaps I misinterpreted your comment but I read it to mean requiring them everywhere (since I view each _expectation_ as a test case, not each _root clause_.

Reply:
I think they should be required for the outermost clause. For the inner clauses, I would recommend them if there is anything that might be non-obvious, but I agree that requiring them for all inner clauses would be too much busywork.